File: 220815/Java/ReverseNaturalNumbers.java
// 문제 URL: https://school.programmers.co.kr/learn/courses/30/lessons/12932?language=java

public class ReverseNaturalNumbers {
  public int[] solution(long givenNumber) {
    int[] reversedDigits = new int[Long.toString(givenNumber).length()];

    for (int i = 0; i < reversedDigits.length; i += 1) {
      reversedDigits[i] = (int) (givenNumber % 10);

      givenNumber /= 10;
    }

    return reversedDigits;
  }
}

File: Previous/220815/JavaScript/reverse-natural-numbers.js
export default function solution(givenNumber) {
  let dividend = givenNumber;

  const array = [];

  let index = 0;
  while (dividend > 0) {
    array[index] = dividend % 10;
    index += 1;
    dividend = parseInt(dividend / 10, 10);
  }

  return array;
}
